src/ion/data_structures.rs

```diff
@@ -203,6 +203,7 @@ pub struct LiveBundle {
     pub allocation: Allocation,
     pub prio: u32, // recomputed after every bulk update
     pub spill_weight_and_props: u32,
+    pub limit: Option<usize>,
```

Comment:
The size of these structs is pretty important for performance (cache residency etc) -- instead of an `Option<usize>` could we use an `Option<u8>` here? (The number of physical registers is already constrained to 128 so this should always fit?)